assets/js/modules/reader-revenue-manager/components/dashboard/PublicationApprovedOverlayNotification.js

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import OverlayNotification from '../../../../components/OverlayNotification/OverlayNotification';
+import ReaderRevenueManagerIntroductoryGraphicDesktop from '../../../../../svg/graphics/reader-revenue-manager-introductory-graphic-desktop.svg';
+import ReaderRevenueManagerIntroductoryGraphicMobile from '../../../../../svg/graphics/reader-revenue-manager-introductory-graphic-mobile.svg';
+import useViewOnly from '../../../../hooks/useViewOnly';
+import useDashboardType from '../../../../hooks/useDashboardType';
+import ExternalIcon from '../../../../../svg/icons/external.svg';
+import { Button } from 'googlesitekit-components';
+import { useSelect, useDispatch } from 'googlesitekit-data';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { CORE_UI } from '../../../../googlesitekit/datastore/ui/constants';
+import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../../../googlesitekit/constants';
+import {
+	MODULES_READER_REVENUE_MANAGER,
+	UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION,
+} from '../../datastore/constants';
+
+export const RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION =
+	'rrmPublicationApprovedOverlayNotification';
+
+export default function PublicationApprovedOverlayNotification() {
+	const isViewOnly = useViewOnly();
+	const dashboardType = useDashboardType();
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const { dismissOverlayNotification } = useDispatch( CORE_UI );
+	const serviceURL = useSelect( ( select ) =>
+		select( MODULES_READER_REVENUE_MANAGER ).getServiceURL()
+	);
+
+	const showApprovedNotificationUI = useSelect( ( select ) =>
+		select( CORE_UI ).getValue(
+			UI_KEY_READER_REVENUE_MANAGER_SHOW_PUBLICATION_APPROVED_NOTIFICATION
+		)
+	);
+
+	/**
+	 * Determines whether the publication approved notification should be shown when the following conditions are met.
+	 * - The notification has not been dismissed.
+	 * - The notification UI is enabled.
+	 * - The user is not in view-only mode.
+	 * - The user is on the main dashboard.
+	 */
+	const shouldShowNotification =
+		isDismissed === false &&
+		showApprovedNotificationUI === true &&
+		! isViewOnly &&
+		dashboardType === VIEW_CONTEXT_MAIN_DASHBOARD;
+
+	const isDismissing = useSelect( ( select ) =>
+		select( CORE_USER ).isDismissingItem(
+			RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const dismissNotification = () => {
+		// Dismiss the notification, which also dismisses it from
+		// the current user's profile with the `dismissItem` action.
+		dismissOverlayNotification(
+			RRM_PUBLICATION_APPROVED_OVERLAY_NOTIFICATION
+		);
+	};
+
+	if ( isDismissing ) {
+		return null;
+	}
```

Comment:
I feel this is redundant and may even have unexpected side effects, for example, if the dismissal request fails for any reason (e.g. a network error), we may see a flash of the notification disappearing and reappearing. We also do not have this check for other overlay notifications. What do you think?

```suggestion
```